kedro/framework/cli/starters.py

```diff
@@ -149,11 +217,32 @@ def starter():
 @starter.command("list")
 def list_starters():
     """List all official project starters available."""
-    repo_url = _STARTERS_REPO.replace("git+", "").replace(".git", "/tree/main/{alias}")
-    output = [
-        {alias: repo_url.format(alias=alias)} for alias in sorted(_STARTER_ALIASES)
-    ]
-    click.echo(yaml.safe_dump(output))
+    starters_aliases = _get_starters_aliases()
+    # we "reverse" the dictionnary so that the origin key in starter config
+    # {name: {template_path; ..., origin: ...}} became the first key:
+    # {kedro: ["astro-iris":{...}], "my-plugin":["my-starter":{...}]}
+    starters_by_origin = {}
+    for starter_name, starter_config in starters_aliases.items():
+        starter_raw_config = {
+            starter_name: {k: v for k, v in starter_config.items() if k != "origin"}
+        }
```

Comment:
You can create a `deepcopy` of `starters_aliases` and use that instead - that way it's safe to use `.pop()` as well, as @AntonyMilneQB suggested.

Reply:
It is still safe here since _get_starters_aliases recreate the dict when it is called and already deepcopy _STARTERS_ALAIASES.